myriad-scheduler/src/main/java/com/ebay/myriad/state/utils/ByteBufferSupport.java

```diff
@@ -0,0 +1,328 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ebay.myriad.state.utils;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.mesos.Protos;
+import com.ebay.myriad.scheduler.NMProfile;
+import com.ebay.myriad.state.NodeTask;
+import com.google.gson.Gson;
+import com.google.protobuf.GeneratedMessage;
+
+/**
+* ByteBuffer support for the Serialization of the StoreContext
+*/
+public class ByteBufferSupport {
```

Comment:
I have no strong opinion on this, but I kind of like all the serialization/deserialization code in one class. But happy to move it to NodeTask.